src/main/java/com/google/devtools/build/lib/syntax/MethodDescriptor.java

```diff
@@ -130,46 +130,51 @@ public Object invoke(Object obj) throws InvocationTargetException, IllegalAccess
   public Object call(Object obj, Object[] args, Location loc, Environment env)
       throws EvalException, InterruptedException {
     Preconditions.checkNotNull(obj);
+    Object result;
     try {
-      Object result = method.invoke(obj, args);
-      if (method.getReturnType().equals(Void.TYPE)) {
-        return Runtime.NONE;
-      }
-      if (result == null) {
-        if (isAllowReturnNones()) {
-          return Runtime.NONE;
-        } else {
-          throw new EvalException(
-              loc,
-              "method invocation returned None, please file a bug report: "
-                  + getName()
-                  + Printer.printAbbreviatedList(ImmutableList.copyOf(args), "(", ", ", ")", null));
-        }
-      }
-      // TODO(bazel-team): get rid of this, by having everyone use the Skylark data structures
-      result = SkylarkType.convertToSkylark(result, method, env);
-      if (result != null && !EvalUtils.isSkylarkAcceptable(result.getClass())) {
-        throw new EvalException(
-            loc,
-            Printer.format(
-                "method '%s' returns an object of invalid type %r", getName(), result.getClass()));
-      }
-      return result;
+      result = method.invoke(obj, args);
     } catch (IllegalAccessException e) {
       // TODO(bazel-team): Print a nice error message. Maybe the method exists
       // and an argument is missing or has the wrong type.
       throw new EvalException(loc, "Method invocation failed: " + e);
-    } catch (InvocationTargetException e) {
-      if (e.getCause() instanceof FuncallExpression.FuncallException) {
-        throw new EvalException(loc, e.getCause().getMessage());
-      } else if (e.getCause() != null) {
-        Throwables.throwIfInstanceOf(e.getCause(), InterruptedException.class);
-        throw new EvalException.EvalExceptionWithJavaCause(loc, e.getCause());
-      } else {
+    } catch (InvocationTargetException x) {
+      Throwable e = x.getCause();
+      if (e == null) {
         // This is unlikely to happen
```

Comment:
Doesn't help for Bazel, and not a new issue, but might be nice to file a BugReport in this case, if we don't expect it to happen.
BugReport.sendBugReport(new IllegalStateException("Unexpected InvocationTargetException with no cause: " + obj + ", " + Arrays.toString(args) + ", " + loc, x));

Or whatever format seems best, like the one below. Inside Google, we'll see any instances of this that happen to Google-internal users.

Reply:
Is it better to explicitly send a bug report here or simply to throw the `IllegalStateException` you suggest? Won't someone at the top of the callchain send a crash report?

Reply:
It's better to send the bug report, because Bazel can recover from this situation: it's an unexpected state, but not an unrecoverable one. I prefer crashing only when there's no way to recover, or correctness is potentially affected.

Reply:
I'm not sure the user experience is much improved in the `BugReport` case. If a user encounters this case, their build will surely fail because Starlark doesn't have exception handling. Perhaps the best case is they get some useful work done because they're using `--keep_going`. But then they'll be faced with what will claim to be a Starlark-level "unexpected method invocation" error, presumably complete with a traceback through `.bzl` files. Maybe the Google `BugReport` implementation somehow informs the user that something strange happened in their build, but a Bazel user will be none the wiser. At least going down in a flood Java tracebacks all tells the user that they're not to blame.

Anyway, I've pushed what I think is best to write for these two cases. If `BugReport` is still the way to go, I'll stop arguing.